js/index.js

```diff
@@ -1,137 +1,192 @@
 document.addEventListener("DOMContentLoaded", function() {
+    // Prevent default refreshing behavior of the "submit" event
+    document.querySelector("form").addEventListener("submit", function(event) {
+        event.preventDefault(); // Prevent the default behavior of form submission
+        // Get the values from the form inputs
+        var usersEmail = document.getElementById("usersEmail").value;
+        var usersName = document.getElementById("usersName").value;
+        var usersMessage = document.getElementById("usersMessage").value;
+
+        // Create a new list item element for a new message
+        var newMessage = document.createElement("li");
+
+        // Set the inner HTML of the newMessage element
+        newMessage.innerHTML = `
+          <a href="mailto:${usersEmail}">${usersName}</a>
+          <span>${usersMessage}</span>
+        `;
+
+        // Create a new button element for editing messages
+        var editButton = document.createElement("button");
+        editButton.innerText = "edit";
+        editButton.type = "button";
+
+        // Add event listener to the editButton
+        editButton.addEventListener("click", function() {
+            var newMessageText = prompt("Enter the new message:");
+            if (newMessageText !== null && newMessageText !== "") {
+                newMessage.querySelector("span").textContent = newMessageText;
+            }
+        });
+
+        // Create a new button element for removing messages
+        var removeButton = document.createElement("button");
+        removeButton.innerText = "remove";
+        removeButton.type = "button";
+
+        // Add event listener to the removeButton
+        removeButton.addEventListener("click", function() {
+            // Find the button's parent element using DOM Traversal
+            var entry = removeButton.parentNode;
+
+            // Remove the entry element from the DOM
+            entry.remove();
+
+            // Update visibility after removing an entry
+            updateVisibility();
+        });
+
+        // Append the editButton and removeButton to the newMessage
+        newMessage.appendChild(editButton);
+        newMessage.appendChild(removeButton);
+
+        // Select the #messages section by id
+        var messageSection = document.getElementById("messages");
+
+        // Select the <ul> element within the messageSection
+        var messageList = messageSection.querySelector("ul");
+
+        // Append the newMessage to the messageList
+        messageList.appendChild(newMessage);
+
+        // Update visibility after adding a new message
+        updateVisibility();
+
+        // Clear form inputs after submission
+        document.getElementById("usersEmail").value = "";
+        document.getElementById("usersName").value = "";
+        document.getElementById("usersMessage").value = "";
+    });
+
+    // Function to update visibility of the messages section
+    function updateVisibility() {
```

Comment:
It's easier to the the hidden attributer on an element rather than modifying the display style and assuming the style is block.